src/providers/spatialite/qgsspatialitesourceselect.cpp

```diff
@@ -312,6 +314,11 @@ bool QgsSpatiaLiteSourceSelect::newConnection( QWidget *parent )
   // inserting this SQLite DB path
   settings.setValue( baseKey + "selected", savedName );
   settings.setValue( baseKey + savedName + "/sqlitepath", myFI.canonicalFilePath() );
```

Comment:
The settings part is no longer required, right? It's already handled in QgsSpatiaLiteProviderConnection::store which will be called by the call to saveConnection below.